utils/calendrical_calculations/src/chinese_based.rs

```diff
@@ -272,17 +264,15 @@ pub(crate) fn new_year_in_sui<C: ChineseBased>(prior_solstice: RataDie) -> (Rata
     let following_solstice = winter_solstice_on_or_before::<C>(prior_solstice + 370); // s2
     let month_after_eleventh = new_moon_on_or_after::<C>((prior_solstice + 1).as_moment()); // m12
     let month_after_twelfth = new_moon_on_or_after::<C>((month_after_eleventh + 1).as_moment()); // m13
+    let month_after_thirteenth = new_moon_on_or_after::<C>((month_after_twelfth + 1).as_moment());
     let next_eleventh_month = new_moon_before::<C>((following_solstice + 1).as_moment()); // next-m11
     let lhs_argument =
         ((next_eleventh_month - month_after_eleventh) as f64 / MEAN_SYNODIC_MONTH).round() as i64;
-    if lhs_argument == 12
-        && (no_major_solar_term::<C>(month_after_eleventh)
-            || no_major_solar_term::<C>(month_after_twelfth))
-    {
-        (
-            new_moon_on_or_after::<C>((month_after_twelfth + 1).as_moment()),
-            following_solstice,
-        )
+    let solar_term_a = major_solar_term_from_fixed::<C>(month_after_eleventh);
+    let solar_term_b = major_solar_term_from_fixed::<C>(month_after_twelfth);
+    let solar_term_c = major_solar_term_from_fixed::<C>(month_after_thirteenth);
+    if lhs_argument == 12 && (solar_term_a == solar_term_b || solar_term_b == solar_term_c) {
```

Comment:
nit: mention in a comment that this is where the inlining of `no_major_solar_term` has happened so it's easier to compare algorithms

Reply:
The inlining impacts the whole function because in addition to inlining `no_major_solar_term`, I'm also pre-calculating `month_after_thirteenth`, which was being calculated previously by `no_major_solar_term` and then again by this function. I'm not sure of the correct way to phrase an additional comment